ConvivaTestApp/src/main/java/com/bitmovin/analytics/conviva/testapp/MainActivity.kt

```diff
@@ -0,0 +1,97 @@
+    var bitmovinPlaybackConfiguration: PlaybackConfiguration? = null
+    var bitmovinAdConfiguration: AdvertisingConfiguration? = null
+    var bitmovinPlayerConfiguration: PlayerConfiguration? = null
+    var bitmovinPlayer: BitmovinPlayer? = null
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        // Get the Intent that started this activity and extract the string
+        val intent = intent
+        val autoPlay = intent.getBooleanExtra(AUTOPLAY_KEY, false)
+        var vmapTagUrl = intent.getStringExtra(VMAP_KEY)
+        bitmovinPlaybackConfiguration = PlaybackConfiguration()
+        bitmovinPlaybackConfiguration!!.isAutoplayEnabled = autoPlay
+
+        // Creating a new PlayerConfiguration
+        bitmovinPlayerConfiguration = PlayerConfiguration()
+        // Assign created SourceConfiguration to the PlayerConfiguration
+        bitmovinPlayerConfiguration!!.sourceConfiguration = bitmovinSourceConfiguration
+        bitmovinPlayerConfiguration!!.playbackConfiguration = bitmovinPlaybackConfiguration
+
+        // Create AdSources
+        if (vmapTagUrl != null) {
+            val vmapAdSource = AdSource(AdSourceType.IMA, vmapTagUrl)
+
+            // Setup ad
+            val vmapAdRoll = AdItem("", vmapAdSource)
+
+            // Add the AdItems to the AdvertisingConfiguration
+            bitmovinAdConfiguration = AdvertisingConfiguration(vmapAdRoll)
+            // Assing the AdvertisingConfiguration to the PlayerConfiguration
+            // All ads in the AdvertisingConfiguration will be scheduled automatically
+            bitmovinPlayerConfiguration!!.advertisingConfiguration = bitmovinAdConfiguration
+        }
+
+        // Create new BitmovinPlayerView with our PlayerConfiguration
+        bitmovinPlayerView = BitmovinPlayerView(this, bitmovinPlayerConfiguration)
+        bitmovinPlayerView!!.layoutParams = LinearLayout.LayoutParams(
+            LinearLayout.LayoutParams.MATCH_PARENT,
+            LinearLayout.LayoutParams.MATCH_PARENT
+        )
+        val rootView = findViewById<View>(R.id.activity_main) as LinearLayout
+
+        // Add BitmovinPlayerView to the layout
+        rootView.addView(bitmovinPlayerView, 0)
+        bitmovinPlayer = bitmovinPlayerView!!.player
+    }
+
+    override fun onStart() {
+        super.onStart()
+        bitmovinPlayerView!!.onStart()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        bitmovinPlayerView!!.onResume()
+    }
+
+    override fun onPause() {
+        bitmovinPlayerView!!.onPause()
+        super.onPause()
+    }
+
+    override fun onStop() {
+        bitmovinPlayerView!!.onStop()
+        super.onStop()
+    }
+
+    override fun onDestroy() {
+        bitmovinPlayerView!!.onDestroy()
+        super.onDestroy()
+    }
+
+    companion object {
+        const val AUTOPLAY_KEY = "autoplay"
+        const val VMAP_KEY = "vmapTag"
+    }
+}
```

Comment:
Please add a newline at EOF

Reply:
Addressed some comments
- Split in to separate files
- Moved all common code in a base class